src/commands/gate/evaluate.ts

```diff
@@ -143,7 +149,42 @@ export class GateEvaluateCommand extends Command {
       })
   }
 
-  private handleEvaluationSuccess(evaluationResponse: EvaluationResponse): number {
+  /**
+   * Evaluate rules with retry
+   * then if service returns status:wait, retry for wait_time_ms
+   * if service returns 5xx, exponential backoff retry
+   */
```

Comment:
it's not quite a "retry". The retry will still be handled by `retryRequest`, but we "fake" a wait time here. It's not very natural so it's worth investing in a longer comment 

```suggestion
  /**
   * Evaluate gate rules and return a promise that will behave as follows:
   * - If the request is successful, the promise will be resolved with the response
   * - If the request is successful but the status is 'wait', the promise will be rejected after the received wait time (wait_time_ms)
   * - If the request is not successful, the promise will be rejected after `initialRetryMs`, with an exponential factor that depends on the attempt (exponential backoff).
   * If the promise is rejected, `retryRequest` will handle the retry immediately.
   */
```